### PR TITLE
Dismissible error message and modify styles for mobile

### PR DESCRIPTION
**Pull request by Dewar0019**
screen shot 2017-03-21 at 10 22 11 am

The iPhone 5 view, however, seems off and was wondering if someone with more CSS experience than I could take a shot at it. 

screen shot 2017-03-21 at 10 23 06 am

### CODE REVIEW

**Comment by grantovich** [comment]
From your screenshots the mobile view seems like it's behaving the same as the desktop view – although one thing I'm noticing in both views is that the "X" icon on the right isn't vertically aligned with the "caution" icon on the left, which looks strange. In the mobile case where the text wraps, we might benefit from having some padding on the left of the "X".

File: client/app/controllers/map.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
+
+  dismissAllStudentsAlert: false,
```
Comment (grantovich) [inline client/app/controllers/map.js]: Naming nitpick: This sort of sounds like a command/function ("dismiss the all-students alert"), when it's meant to be an indication of whether something has occurred ("the all-students alert has been dismissed"). Something like `studentsAlertDismissed` might be clearer.

File: client/app/controllers/map.js [hunk 2]
```diff
@@ -13,7 +15,14 @@ export default Ember.Controller.extend({
   schools: Ember.computed.mapBy('assignedStudents', 'school'),
   uniqueSchools: Ember.computed.uniq('schools'),
 
-  showingAllStudents: Ember.computed('students.@each.isLocated', function() {
-    return this.get('students').every((student) => student.get('isLocated'));
-  })
+  showingAllStudents: Ember.computed('dismissAllStudentsAlert', 'students.@each.isLocated', function() {
```
Comment (grantovich) [inline client/app/controllers/map.js]: The name `showingAllStudents` here is no longer accurate, since the value could be true even if not all students are actually being shown, due to the "dismiss" override. Maybe the logic could be inverted and this could be named `studentsAlertActive`, since that would correspond better with its real purpose anyway.

File: client/app/templates/map.hbs [hunk 3]
```diff
@@ -12,6 +12,7 @@
     {{#unless showingAllStudents}}
       <div class='msg msg--map'>
         {{fa-icon 'warning'}}
+        <i class='fa fa-close close' onClick={{action 'closeAlert'}}></i>
```
Comment (grantovich) [inline client/app/templates/map.hbs]: I believe `<i class='fa fa-close close' {{action 'closeAlert'}}></i>` would be sufficient here, since a click is the default event that triggers an action.

File: config/locales/en.yml [hunk 4]
```diff
@@ -64,7 +64,7 @@ en:
     error:
       generic: |
         Oops! Something went wrong and we couldn't process that action. You
-        might want to try it again, or if that doesn't work, refresh the page.
+        might want to try it again.
```
Comment (grantovich) [inline config/locales/en.yml]: Why the wording change here? Seems unrelated to the PR as a whole.

File: spec/features/user_views_map_spec.rb [hunk 5]
```diff
@@ -89,6 +89,18 @@
     expect(page).to have_content t('map.messages.missingStudents')
   end
 
+  scenario 'missing students alert dismissed' do
```
Comment (grantovich) [inline spec/features/user_views_map_spec.rb]: We've been wording these in a way such that the nesting of the contexts forms a complete sentence when all the descriptions are joined together (as they are in the RSpec output when a scenario fails). So:

```
feature 'User views map' do
  ...
  context 'with multiple bus positions' do
    ...
    scenario 'live-updated as locations change' do
      ...
```

...becomes the valid sentence *"User views map with multiple bus positions live-updated as locations change"*.

So this could be something like `scenario "with a closable warning when not all students are displayed"`.

File: config/locales/en.yml (hunk 4 above)
Comment (Dewar0019) [inline config/locales/en.yml]: This was to accommodate the mobile app where the user wouldn't be able to refresh a page.

File: spec/features/user_views_map_spec.rb (hunk 5 above)
Comment (Dewar0019) [inline spec/features/user_views_map_spec.rb]: 👍

File: client/app/templates/map.hbs (hunk 3 above)
Comment (Dewar0019) [inline client/app/templates/map.hbs]: 👍

File: client/app/controllers/map.js (hunk 1 above)
Comment (Dewar0019) [inline client/app/controllers/map.js]: 👍

File: client/app/controllers/map.js (hunk 2 above)
Comment (Dewar0019) [inline client/app/controllers/map.js]: 👍

File: config/locales/en.yml (hunk 4 above)
Comment (grantovich) [inline config/locales/en.yml]: Ah, I see. That's a bit worrisome, since there might be error conditions where the only way to recover is in fact to refresh the page (I wouldn't expect this to be common, but it's the only 100% effective escape hatch we have). How would we handle this in an app context? Close and re-launch the app?

Comment (Dewar0019) [inline config/locales/en.yml]: The mobile app is essentially a web-view so technically we could refresh the page for the user but I am not sure about what to do with the message or if we should refresh without alerting the user?

Comment (grantovich) [inline config/locales/en.yml]: I don't think we should be auto-refreshing on error. I'm just thinking a refresh should probably always be available, as a manual user-initiated action, somehow. Otherwise there is no escape hatch if our app gets into a bad state that can only be fixed by a complete reset. Maybe this is unnecessarily paranoid, but I have seen some web-view apps that stash a "Reload" option in their ••• menu or somewhere unobtrusive.

Comment (grantovich) [inline config/locales/en.yml]: I agree the wording would need to change regardless, since in an app context "refresh the page" sounds odd (there is no "page", just the app).

Comment (Dewar0019) [inline config/locales/en.yml]: image

I have attached the image just so we have some context. Would it make sense to attach a reload button to the message?

Comment (grantovich) [inline config/locales/en.yml]: I see your point – there's no "app chrome", so no obvious place to put a new control within the app. Maybe it is best to simply remove the "refresh" wording, then. I'll leave the decision to you folks. Like I said, maybe I'm worrying too much about something that will never happen.
